### PR TITLE
GPII-1318: Updated to node 4.x-ready branch of universal

### PR DESCRIPTION
**Pull request by amb26**
Native build now seems reliable so node_modules can be deleted

### CODE REVIEW

**Comment by javihernandez** [comment]
@amb26 sweet!!

**Comment by sgithens** [comment]
@amb26 
I believe the build steps have changed for this refactor correct?  Could you update the README to reflect the changes.

File: gpii/node_modules/registryResolver/index.js [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
 });
 
-fluid.defaults("gpii.lifecycleManager.resolverConfigWithRegistry", {
-    gradeNames: ["autoInit", "fluid.gradeLinkageRecord"],
-    contextGrades: ["gpii.lifecycleManager.standardResolverConfig"],
-    resultGrades: ["gpii.lifecycleManager.registryResolverConfig"]
-});
\ No newline at end of file
+// Contribute the registration of the registryResolver into the LifecycleManager's configuration
+fluid.makeGradeLinkage("gpii.windows.resolverConfigWithRegistry",
```
Comment (kaspermarkus) [inline gpii/node_modules/registryResolver/index.js]: trailing whitespace breaking jshint

File: gpii/node_modules/spiSettingsHandler/test/testSpiSettingsHandler.js [hunk 2]
```diff
@@ -18,16 +18,16 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
...
     $ = fluid.registerNamespace("jQuery");
 
-require("../../../../index.js"); // TODO: proper module system
+require("../src/spiSettingsHandler.js");
 
 jqUnit.module("SpiSettingsHandler Module");
 
 var testPayloads = ["testMouseTrails", "testFilterKeys", "testMouse", "testMouseClickLock", "testLogFont", "testNonClientMetrics", "testStickyKeys", "testHighContrast"];
-
+/*
```
Comment (kaspermarkus) [inline gpii/node_modules/spiSettingsHandler/test/testSpiSettingsHandler.js]: Why are the tests here commented out? Also if they remain commented out, the "testPayloads" definition of line 29 should also be commented out

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+/examples/gpii-oauth2-sample-client/node_modules/
+/examples/gpii-oauth2-sample-client-passport/node_modules/
+/examples/gpii-oauth2-single-process-auth-server/node_modules/
+
+/gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/node_modules/
+/gpii/node_modules/gpii-oauth2/gpii-oauth2-utilities/node_modules/
+
+npm-debug.log
+*.bak
```
Comment (kaspermarkus) [inline .gitignore]: This should be fixed to match what we want ignored in windows

File: gpii/node_modules/spiSettingsHandler/test/testSpiSettingsHandler.js (hunk 2 above)
Comment (amb26) [inline gpii/node_modules/spiSettingsHandler/test/testSpiSettingsHandler.js]: Pure blunder... I was just trying to do some light testing on my machine without destroying its desktop

**Comment by amb26** [comment]
Updated and working with the dynamic device reporter with corresponding commits in https://github.com/GPII/universal/pull/425. Incorporated material from https://github.com/GPII/windows/pull/66/files with fixes, and fixed broken tests from https://github.com/GPII/windows/pull/59

**Pull request merged**